zcash_client_backend/src/proto.rs

```diff
@@ -299,6 +299,15 @@ impl service::TreeState {
             self.height
                 .try_into()
                 .map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "Invalid block height"))?,
+            BlockHash::try_from_slice(&hex::decode(&self.hash).map_err(|e| {
```

Comment:
Because `self.hash` is a protobuf `string`, not a `bytes`, it means it is directly conveying the hex encoding from `zcashd`'s JSON-RPC. Hence it needs to be byte-reversed as part of being parsed (unlike `CompactBlock` where I handled this for the caller).